schema.json

```diff
@@ -0,0 +1,263 @@
+{
+	"$schema": "http://json-schema.org/draft-07/schema#",
+	"default": {
+		"edges": [],
+		"nodes": []
+	},
+	"properties": {
+		"required": true,
+		"edges": {
+			"items": {
+				"$ref": "#/definitions/JSONCanvasEdge"
+			},
+			"type": "array"
+		},
+		"nodes": {
+			"items": {
+				"$ref": "#/definitions/JSONCanvasNode"
+			},
+			"type": "array"
+		}
+	},
+	"definitions": {
+		"JSONCanvasColor": {
+			"anyOf": [
+				{
+					"type": "string"
+				},
+				{
+					"$ref": "#/definitions/JSONCanvasColorPreset"
+				}
+			]
+		},
+		"JSONCanvasColorPreset": {
+			"enum": [1, 2, 3, 4, 5, 6],
+			"type": "number"
+		},
+		"JSONCanvasEdge": {
+			"additionalProperties": false,
+			"properties": {
+				"color": {
+					"$ref": "#/definitions/JSONCanvasColor"
+				},
+				"fromNode": {
+					"type": "string"
+				},
+				"fromSide": {
+					"$ref": "#/definitions/JSONCanvasEdgeSide"
+				},
+				"id": {
+					"type": "string"
+				},
```

Comment:
Define a `NodeId` schema. That will help to give semantic info to properties like `fromNode`.